### PR TITLE
[typescript] Fix uncompilable output when using allOf (#8000)

### LINKED ISSUE #8000: [BUG] [typescript-angular] AllOf hierarchy produces api that doesn't compile
**OpenAPI Spec**
```
openapi: 3.0.3

info:
  title: Demo
  version: 1.0.0
  description: Demo

paths:
  /some-path:
    post:
      operationId: create
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/ExtendedTransferObjectRequest'
      responses:
        201:
          description: some description
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ExtendedTransferObjectResponse'

components:
  schemas:
    TransferObject:
      type: object
      properties:
        id:
          type: string
          format: uuid
    ExtendedTransferObject:
      type: object
      allOf:
        - $ref: '#/components/schemas/TransferObject'
        - type: object
      required:
        - some-field
      properties:
        some-field:
          type: string
    ExtendedTransferObjectRequest:
      type: object
      allOf:
        - $ref: '#/components/schemas/ExtendedTransferObject'
        - type: object
    ExtendedTransferObjectResponse:
      type: object
      allOf:
        - $ref: '#/components/schemas/ExtendedTransferObject'
        - type: object
```

**Generated with**
```
docker run --rm -v "${PWD}:/local" openapitools/openapi-generator-cli:latest generate -i /local/sample.yml -g typescript-angular -o /local/build
```

**Results in a `default.service.ts` file with a invalid import statement along with unused class `ExtendedTransferObject`:**
```
import { ExtendedTransferObject &amp; object } from '../model/models';
```

### PR DESCRIPTION
**Pull request by ralphpig**
Fixes #8000 

# Summary
When using `allOf`, or `oneOf` with an inline schema, a Union/Intersection type is generated with the primitive `object`. This results in an invalid import because the composed type of the schema is ignored.

```js
import { BaseObject &amp; object }
```

# Approach
To fix this I used  `needToImport` to removes the language primitives from being imported. This same check already happens in `addImport`, but not in `toModelImportMap`.

`toModelImportMap` splits union types but not intersection types, so I've combined those cases into `splitComposedType`. For `splitComposedType` I combined `replace(" ","").split("\\|")` and `split("( [|&] )|[<>]")` into `replace(" ","").split("[|&<>]")`. These should be equivilant regex, and I think stripping the whitespace is good here.

I added the warning in `toModelImport` strictly because there was already a warning for the union type and wanted to be consistent.

# Minimal Reproducible Example

```
-g typescript-axios --additional-properties=withSeparateModelsAndApi=true,apiPackage=api,modelPackage=models
```
```yaml
openapi: 3.0.3

info:
  title: Issue 8000
  version: 1.0.0
  description: Issue 8000 Minimal Reproducible Example

paths:
  /resource:
    get:
      operationId: get_resource
      responses:
        200:
          description: resource
          content:
            application/json:
              schema:
                type: object
                allOf:
                  - $ref: '#/components/schemas/BaseObject'
                  - type: object
                    properties:
                      field:
                        type: string

components:
  schemas:
    BaseObject:
      type: object
      properties:
        id:
          type: integer
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
  - No changes
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

### CODE REVIEW

**Comment by ralphpig** [comment]
Hey, just a gentle bump. If anyone has a few minutes to check this out I would greatly appreciate it.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/SharedTypeScriptTest.java [hunk 1]
```diff
@@ -69,4 +71,25 @@ public void oldImportsStillPresentTest() throws IOException {
 
         FileUtils.deleteDirectory(new File("src/test/resources/oldImportsStillPresentTest/"));
     }
+
+    /*
+        #8000
+        Test that primatives are not returned by toModelImportMap
```
Comment (amakhrov) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/SharedTypeScriptTest.java:77]: the comment is a bit too narrow - the actual test not only verifies that, but also checks that  intersection/union type imports are properly handled in general. maybe reword this somehow?

**Review by amakhrov — APPROVED** [review]
Thanks for your contribution! The fix looks good to me.
